include/ros1_bridge/factory.hpp

```diff
@@ -152,6 +152,92 @@ class Factory : public FactoryInterface
     ROS1_T & ros1_msg);
 };
 
+template <class R1_TYPE, class R1_REQ, class R1_RES, class R2_TYPE, class R2_REQ, class R2_RES>
+class ServiceFactory : public ServiceFactoryInterface
+{
+private:
+
+  void translate_1_to_2(const R1_REQ&, R2_REQ&);
+  void translate_1_to_2(const R1_RES&, R2_RES&);
```

Comment:
They do reuse those generated functions. They call convert*() functions for all non-primitive data types.

If you mean, that we could use that template generating convert*() to generate functions converting requests and responses, it is possible but not that trivial. I would need to rewrite this python script and I do not have time to do it now.